pliers/extractors/text.py

```diff
@@ -402,3 +402,33 @@ def _extract(self, stim):
 
         return ExtractorResult(features_list, stim, self,
                                features=self.features, orders=order_list)
+
+class WordCounterExtractor(TextExtractor):
+
+    ''' Extracts number of times word has occurred within text
+    
+    Args:
+        lemmatize(bool): specifies if words are to be lemmatized before being counted. 
+        '''
+
+    def __init__(self, lemmatize=None):
+        super(WordCounterExtractor, self).__init__()
+        self.lemmatize = lemmatize
+
+    def _extract(self, stim):
+        
+        
+        tokens = nltk.word_tokenize(stim.text)
+        
+        if self.lemmatize:
+            lemmatizer=nltk.WordNetLemmatizer()
+            pos_map = dict(zip(['ADJ', 'ADJ_SAT', 'ADV', 'NOUN', 'VERB'], ['a', 's', 'r', 'n', 'v'])) # map tags to wordnet tagset
+            tokens = dict(nltk.pos_tag(tokens, tagset='universal'))
+            tokens = {k: pos_map[v] if v in pos_map else 'n' for k, v in tokens.items()}
+            tokens = [lemmatizer.lemmatize(k, pos=v) for k, v in tokens.items()]
+            
+        word_counter = pd.Series(tokens).groupby(tokens).cumcount()
```

Comment:
Could also use `pd.Series(tokens).value_counts()`—will produce the same result, but I suspect this version might be slightly faster (though I doubt there's a meaningful difference).

Reply:
It seems like value_counts returns summary value of number of occurrences for each unique word, rather than the cumulative count for each token in the sequence.